src/main/java/org/broadinstitute/hellbender/tools/variantdb/nextgen/ExtractFeaturesEngine.java

```diff
@@ -129,18 +136,62 @@ public void traverse() {
                                                                                              INDEL_QUAL_THRESHOLD);
 
         final String userDefinedFunctions = ExtractFeaturesBQ.getVQSRFeatureExtractUserDefinedFunctionsString();
+        Map<String, String> labelForQuery = createQueryLabels(queryLabels);
 
         final StorageAPIAvroReader storageAPIAvroReader = BigQueryUtils.executeQueryWithStorageAPI(
                 featureQueryString,
                 SchemaUtils.FEATURE_EXTRACT_FIELDS,
                 projectID,
                 userDefinedFunctions,
                 useBatchQueries,
-                null);
+                labelForQuery);
 
         createVQSRInputFromTableResult(storageAPIAvroReader);
     }
 
+    private Map<String, String>  createQueryLabels(List<String> labelStringList) {
```

Comment:
This is a great candidate for some unit tests...

Reply:
do I just want to unit test createQueryLabels? or ExtractFeaturesEngine?